### PR TITLE
Update cargo version

### PR DESCRIPTION
**Pull request by LucasSte**
Besides bumping the cargo version, I updated the compiler-builtins version.

### CODE REVIEW

File: crates/cargo-test-macro/src/lib.rs [hunk 1]
```diff
@@ -187,7 +187,7 @@ fn to_token_stream(code: &str) -> TokenStream {
 
 static mut VERSION: (u32, bool) = (0, false);
 
-fn version() -> &'static (u32, bool) {
+fn version() -> (u32, bool) {
```
Comment (dmakarov) [inline crates/cargo-test-macro/src/lib.rs]: This seems to be unrelated to our changes. Why did this change become necessary?

File: tests/build-std/main.rs [hunk 2]
```diff
@@ -153,7 +159,7 @@ fn cross_custom() {
             r#"
             {
                 "llvm-target": "x86_64-unknown-none-gnu",
-                "data-layout": "e-m:e-i64:64-f80:128-n8:16:32:64-S128",
+                "data-layout": "e-m:e-p270:32:32-p271:32:32-p272:64:64-i64:64-f80:128-n8:16:32:64-S128",
                 "arch": "x86_64",
```
Comment (dmakarov) [inline tests/build-std/main.rs]: Similarly, unrelated change to compiler-builtins version patching.

File: crates/cargo-test-macro/src/lib.rs (hunk 1 above)
Comment (LucasSte) [reply crates/cargo-test-macro/src/lib.rs]: The CI fetches the latest version of cargo nightly and cargo beta, which error out in these lines. When the version 1.75.0 was released, this was not an issue with the nightly or beta, but it is today.

Comment (LucasSte) [reply crates/cargo-test-macro/src/lib.rs]: I removed the changes, but the CI will fail. Should we disable rust nightly and beta tests?

File: crates/xtask-bump-check/src/xtask.rs [hunk 3]
```diff
@@ -158,15 +158,16 @@ fn bump_check(args: &clap::ArgMatches, config: &cargo::util::Config) -> CargoRes
...
+    // This test does not work because we are running it against a too old version of cargo.
+    // if let Some(referenced_commit) = referenced_commit.as_ref() {
+    //     let mut cmd = ProcessBuilder::new("cargo");
+    //     cmd.arg("semver-checks")
+    //         .arg("--workspace")
+    //         .arg("--baseline-rev")
+    //         .arg(referenced_commit.id().to_string());
+    //     config.shell().status("Running", &cmd)?;
+    //     cmd.exec()?;
+    // }
```
Comment (dmakarov) [inline crates/xtask-bump-check/src/xtask.rs]: Can this be made less 'intrusive' using a `#[cfg()]` annotation? Maybe something like
```
#[cfg(target_os = "solana")]
```
before the `if-`statement? Since such a `target_os` doesn't exist for host compilation, the following `if-`statement will always be excluded from compiled code.

**Review by dmakarov — APPROVED** [review]
